spec/rspec/support/object_inspector_spec.rb

```diff
@@ -92,6 +92,17 @@ def with_date_loaded
             matcher_without_a_description.inspect)
         end
       end
+
+      context "with double objects" do
+        it "displays double name" do
+          double = double('Foo')
+          allow(double).to receive(:name).and_return("Foo")
+          ObjectInspector.register double.class do |double|
+            "Double (#{double.name})"
+          end
+          expect(ObjectInspector.inspect(double)).to eq("Double (Foo)")
+        end
+      end
```

Comment:
This spec should be removed.  As I tried to explain above, it's problematic for a few reasons and is really unnecessary since the implementation is blissfully unaware of rspec-mocks, test doubles, or any other such thing.

Instead, we'll wind up having some specs in rspec-mocks for this kind of thing once we do a PR there that builds on this one.